### ISSUE TITLE
Tensorflow model

### ISSUE THREAD

**Issue by WOLFI3654** [WOLFI3654/WaveAnalyzing#1]
It would be very nice to have a template to work with. I'm new to python

**Comment by GBLin5566** [comment]
Hi, since you are new to Python, I'll recommend to use Keras as your framework.
Keras is a high-level api framework of lots of ML backend(including Tensorflow).
This is a simple tutorial of using Keras to build RNN model (which might be suitable in your case).
https://github.com/fchollet/keras/blob/master/examples/imdb_lstm.py

**Comment by WOLFI3654** [comment]
@GBLin5566 Thats very interesting. I've heard about Keras.
Is the given example something that would work for my Problem?

**Comment by GBLin5566** [comment]
According to the description of README, it looks like you want to input time-series data, and then predict the binary classification result. 
Recurrent neural network (RNN) is well-known for its ability to model time-series data. I'll work on a simple template and add some comments.

**Comment by WOLFI3654** [comment]
Oh thats great. Thank you <3

**Comment by WOLFI3654** [comment]
@GBLin5566 I think Keras is neither using Tensorflow nor my gpu.
GPU load is around 1% and Keras will only run when Thenao is installed

**Comment by WOLFI3654** [comment]
**Test score: 0.716750392318
Test accuracy: 0.4925**
Looks not very good...

**Comment by GBLin5566** [comment]
@WOLFI3654 Check this out for using tf as backend and also how to use gpu.
https://keras.io/backend/

The training data & testing data are both made-up now, so it is obvious why the acc. are around 0.5. (The acc. of random guess).
You should fit your own data into 'change the code here' block.

**Comment by WOLFI3654** [comment]
Okay. And how can I save the model to process my data later

**Comment by WOLFI3654** [comment]
I think I could use 

`model.save_weights(filepath)`

at the end and when I want to Train I just load the weights instead of Training it...?


[Like this?](https://github.com/WOLFI3654/WaveAnalyzing/commit/68ac92f6b7f396b1d48c3c5eb75e0152b90efaac)

**Comment by GBLin5566** [comment]
Hi @WOLFI3654 , sorry for the late replying.
I've comment in the file, please check for it.

**Comment by WOLFI3654** [comment]
@GBLin5566 I've commented something under your comment a while ago

**Comment by WOLFI3654** [comment]
"how can I feed the Network with a continuos data flow?
I think something like an async call or something is needed to still print the Output in real time"

**Comment by GBLin5566** [comment]
So you want to feed in the testing data from somewhere remotely (when ever you want) and output the result ?
In this case, you might need to build a server with a trained model.
When the server get the http request(POST or whatever), the trained model inference the results and return as a response.

If you simply want to run it locally, then feed in  your data with a python for loop is enough.

**Comment by WOLFI3654** [comment]
I think you got it wrong.
I have data which is similar to an audio sample. The data is local but each amount of time I can feed in one data Point. So the Network should get more sure over time what he is being feed with. And I want somehow to see the answer in realtime

**Comment by GBLin5566** [comment]
Then I think using buffer to store the data might be a good practice.
For example, buffer 1 stores data point data_1, data_2, ..., data_8 and feed them into nn to get the output of this period. Buffer 2 stores data point data_2, data_3, ..., data_9 and feed them, and keep doing the same process. In this case, you might do need some async functions to store the data into buffers.

**Comment by WOLFI3654** [comment]
It's not like a few seconds of data can determine what it is being feed with

**Comment by GBLin5566** [comment]
This might be a good ref for your problem.
https://www.reddit.com/r/MachineLearning/comments/7axg9w/d_how_to_train_your_network_on_streaming_data/

**Comment by WOLFI3654** [comment]
So should I collect a few data over a few seconds and then check them?
